src/source.ts

```diff
@@ -138,12 +140,24 @@ export class Source<T extends IAnyModelType, Q extends Query>
     ) {
       return result.values;
     }
-    const items = await this._load(q);
+    const items = await this.loadOrExisting(key, q).finally(() =>
+      this._existingLoad.delete(key)
+    );
     const values = items.map((item: any) => this.addOrUpdate(item));
     this.setCache(key, values, timestamp);
     return values;
   }
 
+  async loadOrExisting(key: string, q: Q): Promise<SnapshotIn<T>[]> {
+    const existingLoad = this._existingLoad.get(key);
+    if (existingLoad) {
+      return existingLoad;
+    }
+    const load = this._load(q);
```

Comment:
I would call this 'promise' or 'loadPromise'. Calling it 'load' is confusing as it looks like a load function.